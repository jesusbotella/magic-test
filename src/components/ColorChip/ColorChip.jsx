import React from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  chip: {
    borderRadius: 50,
    borderColor: '#e3e3e3',
    borderWidth: 2,
    width: 10,
    height: 10
  }
});

export default function ColorChip(props) {
  const { color } = props;

  const style = {
    ...styles.chip,
    // Let's avoid mapping because all colors
    // are available as CSS color strings
    backgroundColor: color.toLowerCase()
  };

  return <View style={style} />;
}

ColorChip.propTypes = {
  color: PropTypes.string
};

ColorChip.defaultProps = {
  color: ''
};
