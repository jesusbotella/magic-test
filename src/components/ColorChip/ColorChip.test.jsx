import React from 'react';
import { render } from 'react-native-testing-library';
import ColorChip from './ColorChip';

describe('ColorChip', () => {
  it('should render properly', () => {
    const element = render(<ColorChip color="green" />).toJSON();
    expect(element).toMatchSnapshot();
  });
});
