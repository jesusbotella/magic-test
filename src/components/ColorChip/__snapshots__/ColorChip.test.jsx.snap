// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorChip should render properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
      "borderColor": "#e3e3e3",
      "borderRadius": 50,
      "borderWidth": 2,
      "height": 10,
      "width": 10,
    }
  }
/>
`;
